pkg/security/podsecuritypolicy/provider.go

```diff
@@ -239,7 +239,8 @@ func (s *simpleProvider) ValidatePodSecurityContext(pod *api.Pod, fldPath *field
 
 	// TODO(tallclair): ValidatePodSecurityContext should be renamed to ValidatePod since its scope
 	// is not limited to the PodSecurityContext.
-	if len(pod.Spec.Volumes) > 0 && !psputil.PSPAllowsAllVolumes(s.psp) {
+	if len(pod.Spec.Volumes) > 0 {
+		allowsAllVolumes := psputil.PSPAllowsAllVolumes(s.psp)
```

Comment:
nit: rename to `allowsAllVolumeTypes` since this check doesn't allow all volumes anymore.